### DISCUSSION TITLE
Support for SNI smtp certificates

### DISCUSSION THREAD

**Discussion by arthurzenika** [postalserver/postal#2656] [Feature suggestions]
We have a use case where postal is used for receiving email from the internet, posting that message to an application (http endpoints) and then the application can send email back through postal. When activating the SMTP SSL certificates we're failing to connect over ssl from the application to the postal SMTP because it has two different domain names : subdomain-in-mx.example.org (on the public side) and postal-server.internal.example.int (on the internal side). We would like to avoid having the connexion go through the internet and then back into the postal server again (hair-pinning etc.) 

It seems that SNI is technically possible as described in the postfix TLS docs : http://www.postfix.org/TLS_README.html

We haven't found anything in the documentation about multiple SSL certificates, but maybe we missed it. And if this scenario is not yet possible in postal, would this feature interest other users ?

**Comment by willpower232** [comment]
I can't say I've experienced anything validating that the SMTP certificate matches the connected hostname, what software are you using?

**Reply by arthurzenika** [reply]
The application is a java application using [javamail in java spring boot](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/mail/javamail/JavaMailSenderImpl.html),  see [variables you can define](https://javaee.github.io/javamail/docs/api/com/sun/mail/smtp/package-summary.html) or [stackoverflow explanations](https://stackoverflow.com/questions/47166425/how-to-force-javamailsenderimpl-to-use-tls1-2#47720397)...

Right now our workaround is to trust any certificate which encrypts the communication but opens up to man-in-the-middle risks... 

Part of the traceback it displays when failing to validate the certificate looks like this : https://gist.github.com/arthurzenika/ec5f0655fc345fc9f945f0b0109c7179

**Reply by willpower232** [reply]
> unable to find valid certification path to requested target

if you do use the public facing side of things, does your code work or do you get the same error message?

**Reply by arthurzenika** [reply]
Using the public facing FQDN works yes, but the application has to run in an environment which doesn't have access to public ressources for security reasons (mainly).

**Reply by willpower232** [reply]
thats fine, I just wondered if it worked at all :sweat_smile: 

technically you could run a separate docker container for the smtp server with its own SSL certificate only for your internal domain looking at the same database and rabbit, it would be good to confirm that that worked in case your code is struggling to verify certificates for whatever internal domain name you're using
